cfn-resources/organization/cmd/resource/resource.go

```diff
@@ -0,0 +1,365 @@
+		Message:         "Update Completed",
+		ResourceModel:   currentModel}, nil
+}
+
+// Delete handles the Delete event from the Cloudformation service.
+func Delete(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup()
+
+	modelValidation := validator.ValidateModel(DeleteRequiredFields, currentModel)
+	if modelValidation != nil {
+		return *modelValidation, nil
+	}
+
+	atlas, peErr := util.NewAtlasV2OnlyClient(&req, currentModel.AwsSecretName, false)
+	if peErr != nil {
+		return *peErr, nil
+	}
+
+	// Callback
+	if _, idExists := req.CallbackContext[constants.StateName]; idExists {
+		return deleteCallback(atlas, currentModel)
+	}
+
+	// Read before delete
+	_, response, err := currentModel.getOrgDetails(atlas, currentModel)
+	defer closeResponse(response)
+	if err != nil {
+		return handleError(response, constants.DELETE, err)
+	}
+
+	// If exists
+	_, response, err = currentModel.getOrgDetails(atlas, currentModel)
+	defer closeResponse(response)
+	if err != nil && response.StatusCode == http.StatusUnauthorized {
+		return handleError(response, constants.DELETE, err)
+	}
+
+	deleteRequest := atlas.AtlasV2.OrganizationsApi.DeleteOrganization(
+		context.Background(),
+		*currentModel.OrgId,
+	)
+
+	// Since the Delete API is synchronous and takes more than 1 minute most of the time,
+	// we need to make the call in a goroutine and return a progress event
+	// after 10 Seconds. Reason for wait is that the Delete API
+	// may throw error immediately if the resource is not found.
+
+	responseChan := make(chan DeleteResponse, 1)
+	go func() {
+		_, response, err := deleteRequest.Execute()
+		defer closeResponse(response)
+		responseChan <- DeleteResponse{Error: err, Response: response}
+	}()
+
+	currentModel.IsDeleted = util.Pointer(false)
+	select {
+	case responseMsg := <-responseChan:
+		if responseMsg.Error != nil {
+			return handleError(responseMsg.Response, constants.DELETE, responseMsg.Error)
+		}
+
+	case <-time.After(30 * time.Second):
+		// If the Delete is not completed in the above time,
+		// we return a progress event with inProgress status and callback context
+		return handler.ProgressEvent{
+			OperationStatus:      handler.InProgress,
+			Message:              DeleteInProgress,
+			ResourceModel:        currentModel,
+			CallbackDelaySeconds: CallBackSeconds,
+			CallbackContext: map[string]interface{}{
+				constants.StateName: DeletingState,
+			},
+		}, nil
+	}
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         DeleteCompleted,
+		ResourceModel:   nil}, nil
+}
+
+func deleteCallback(atlas *util.MongoDBClient, currentModel *Model) (handler.ProgressEvent, error) {
+	// Read before delete
+	org, response, err := currentModel.getOrgDetails(atlas, currentModel)
+	defer closeResponse(response)
+	if err != nil {
+		if response.StatusCode == http.StatusUnauthorized {
+			return handler.ProgressEvent{
+				OperationStatus: handler.Success,
```

Comment:
why is this success?

Reply:
when the Org is deleted, the DELETE API is responding with Unauthorized instead 404 Error.

Reply:
Jut to confirm: the DELETE API is the ATLAS Delete API? Through the SDK?

Reply:
@marcosuma : DELETE call is through SDK